assets/js/truelayer-for-woocommerce.js

```diff
@@ -0,0 +1,55 @@
+/* global truelayerParams */
+jQuery(function ($) {
+    var truelayerForWooCommerce = {
+        Truelayer: null,
+
+        Payment: null,
+
+        init: function () {
+            window.addEventListener("hashchange", truelayerForWooCommerce.handleHashChange);
+            this.Truelayer = window.Truelayer;
+            this.Payment = this.Truelayer?.Payment;
+        },
+
+        handleHashChange: function() {
+            var currentHash = location.hash;
+            var splittedHash = currentHash.split(":");
+            if( splittedHash[0] === "#truelayer" ){
+                var url = atob( splittedHash[1] );
+                var payment_id = atob( splittedHash[2] );
+                var payment_token = atob( splittedHash[3] );
+                truelayerForWooCommerce.initializeEpp(url, payment_id, payment_token);
+            }
+        },
+
+        initializeEpp: function (url, payment_id, payment_token) {
+            if (!this.Payment) {
+                console.log("Truelayer Payment is not initialized");
+                return;
+            }
+
+            const payment = this.Payment({
+                payment_id: payment_id,
+                resource_token: payment_token,
+                return_uri: url,
+                production: truelayerParams.production,
+                onHandoffStart: () => {
```

Comment:
Why do we redirect away here? This callback is invoked when the user scans the QR code on the HPP to continue the payment authorisation on mobile. The EPP will handle this case by showing a spinner while the user authorises the payment and then redirecting back when ready.

Reply:
I think this might have been just a misunderstanding of the implementation. It has been removed in the last commits